### ISSUE TITLE
Rename/reorganize repo

### ISSUE THREAD

**Issue by lintool** [castorini/SolrAnserini#10]
@r-clancy for discussion:

Should we rename repo to `Anserini-Solr` to be consistent with `Anserini-Spark` and other repos?

Currently, this repo is both Solr + Docker all rolled up into one. Is this what we actually want? Alternative I can think of include:

1. Solr-specific stuff in a separate repo, which yet another Docker repo pulls. This way if we set up Solr in a different manner (e.g., on himrod) we can reuse some of the data.
2. Solr-specific stuff directly into the Anserini repo.

In other words, we have three distinct abstractions:

a. Anserini core
b. code bits to make Anserini work with Solr
c. Docker for convenience

Currently, b+c are together. With option 1 option, all three are in separate repos; downside is proliferation of small repos to manage. With option 2, a+b are together; downside is bloat in the core Anserini codebase.

Thoughts?

**Comment by ryan-clancy** [comment]
Maybe we should re-name it to `Anserini-Docker` as a general catch-all for Docker stuff? We can create a `solr` directory and have stuff live in there... we can Docker-ize more stuff in the same repo if needed in the future too.

I generally like to keep stuff separate, so I'd be fine with having a few different repos:
- `Anserini`
- `Anserini-Spark`
- `Anserini-Solr`
- `Anserini-Docker`

Downside being we'd need to keep versions up-to-date, etc... Would merging `Anserini`, `Anserini-Spark`, and `Anserini-Solr` keeping everything in their own package be a better middle ground? That way we keep all the code together, but still have the Docker stuff in a separate repo.

@lintool

**Comment by lintool** [comment]
> Downside being we'd need to keep versions up-to-date, etc... Would merging Anserini, Anserini-Spark, and Anserini-Solr keeping everything in their own package be a better middle ground? That way we keep all the code together, but still have the Docker stuff in a separate repo.

The downside there is fatjar bloat. There's also `Anserini-embeddings` - see thread in https://github.com/castorini/Anserini/issues/485

If we roll everything together, the fatjar will probably approach 1g... which will... be... slow...

`Anserini-Solr` and `Anserini` together might be a reasonable compromise though, since we've actually gone down this route already? https://github.com/castorini/Anserini/pull/443

So, under this plan, we can have setup as follows:

```
+
|- Anserini-Docker
|- Anserini
```
And `Anserini-Docker` can reach inside `Anserini` (i.e., using relative path) to grab all the data/code it needs.

Add  @Peilin-Yang if he has any comments...

**Comment by ryan-clancy** [comment]
Ahh, I forgot about that change.

So we would have:
- `Anserini` (core + Solr)
- `Anserini-Spark`
- `Anserini-Docker`
- `Anserini-embeddings`

**Comment by lintool** [comment]
Let's go with this tentatively... but we should let folks actually enjoying the holidays at least a bit a time to have a say... :)

🎄 🎁 ☃️

**Comment by ryan-clancy** [comment]
Closing as this has been folded into the main anserini repo...
